Source/missiles.cpp

```diff
@@ -1928,13 +1923,9 @@ void AddLightball(Missile &missile, Point dst, Direction /*midir*/)
 	UpdateMissileVelocity(missile, dst, 16);
 	missile._miAnimFrame = GenerateRnd(8) + 1;
 	missile._mirange = 255;
-	if (missile._misource < 0) {
-		missile.var1 = missile.position.start.x;
-		missile.var2 = missile.position.start.y;
-	} else {
-		missile.var1 = Players[missile._misource].position.tile.x;
-		missile.var2 = Players[missile._misource].position.tile.y;
-	}
+	Point position = missile._misource < 0 ? missile.position.start : Players[missile._misource].position.tile;

```

Comment:
Consider marking `const` and using initializer list:
```suggestion
	const Point position { missile._misource < 0 ? missile.position.start : Players[missile._misource].position.tile };
```